### PR TITLE
Improved the way the upload procedure reports progress.

### PR DESCRIPTION
**Pull request by spherefoundry**
# Checklist:

- [x] Code follows the style guidelines
- [x] Documentation updated
- [x] Corner cases evaluated and handled
- [x] Automatic tests added/update
- [x] Automatic tests pass
- [x] Software runs (locally or hosted)
- [x] Sanity check passed
- [x] Self-review passed

### CODE REVIEW

File: src/services/data_model_engine.js [hunk 1]
```diff
@@ -213,23 +213,19 @@ export default class DataModelEngine {
     await this.entityRepository.discardBundling(bundleStubId);
   }
 
-  async uploadAcceptedBundleCandidates() {
+  async uploadAcceptedBundleCandidates(callbacks = {success: async () => {}, fail: async () => {}}) {
```
Comment (vanruch) [inline src/services/data_model_engine.js]: I don't think it's a good idea to create new callbacks as a default parameter. Usually it's handled in a way that if callback is defined => call it

Comment (vanruch) [reply src/services/data_model_engine.js]: Or maybe at least use Function() constructor?

Comment (krzysztof-jelski) [inline src/services/data_model_engine.js]: Suggestion: rename `callbacks` to `uploadProgressCallbacks`

Comment (spherefoundry) [reply src/services/data_model_engine.js]: I like the way it is right now as it shows that the functions can be `async`. Even better would be to use TypeScript 😮

File: src/services/data_model_engine.js [hunk 2]
```diff
@@ -213,23 +213,19 @@ export default class DataModelEngine {
...
     const waitingBundles = await this.bundleRepository.findBundlesWaitingForUpload();
```
[the remark this answers is not in the document]
Comment (spherefoundry) [reply src/services/data_model_engine.js]: We also need storagePeriods :D I have modified the local variables a bit, but reducing the return values is IMO an overkill.

File: src/services/data_model_engine.js [hunk 3]
```diff
@@ -213,23 +213,19 @@ export default class DataModelEngine {
...
-    const summary = {
-      ok: {},
-      failed: {}
-    };
     for (const waitingBundle of waitingBundles) {
```
[the remark this answers is not in the document]
Comment (spherefoundry) [reply src/services/data_model_engine.js]: ✅

File: src/services/data_model_engine.js (hunk 1 above)
Comment (spherefoundry) [reply src/services/data_model_engine.js]: ✅
